src/main/java/com/instaclick/pentaho/plugin/amqp/AMQPPlugin.java

```diff
@@ -515,10 +686,37 @@ public void dispose(StepMetaInterface smi, StepDataInterface sdi)
         meta = (AMQPPluginMeta) smi;
         data = (AMQPPluginData) sdi;
 
+        logDebug("DISPOSE INVOKED");
+
+        waitForConfirtmationStepsFinished();

```

Comment:
Why do we need to wait here and  keep track of ``watchedConfirmStep`` ?
Can we just delay the flush until the ``transListener`` gets invoked ?

```java
if (data.isTransactional || data.activeConfirmation) {
    getTrans().addTransListener(transListener);
}


// ....


if ( ! data.isTransactional && ! data.activeConfirmation) {
    flush();
}
```